tensilelite/Tensile/KernelWriter.py

```diff
@@ -2330,7 +2330,8 @@ def _loopBody( self, kernel, tensorParametersA, tensorParametersB, pack, lc, loo
       module.addComment2("Unrolled Loop %u/%u - Begin" % (lc+1, loopCopies))
     if kernel["PrefetchGlobalRead"] and not self.states.numItersPLR and not kernel["ScheduleIterAlg"] == 2:
       if kernel["DirectToLdsA"] or kernel["DirectToLdsB"]:
-        module.add(self._wait(kernel, tensorParametersA, tensorParametersB, 0, -1, -1, "11wait for global read"))
+        vmcntVal = 1 if kernel["PrefetchGlobalRead"] == 2 else 0
+        module.add(self._wait(kernel, tensorParametersA, tensorParametersB, vmcntVal, -1, -1, "11wait for global read"))
```

Comment:
You can apply the same logic to NoLoadLoop (for NGLL).
Another thing is that you can skip the local write wait below for DTL (you can use "NoLdsWriteCode" for check).

I did not care much about PLR=0 case.
There might be some more chances for optimizations for PLR=0.
By the way, I think it is better to change the title (PLR=0 instead of DU=MI-K).

Reply:
Thanks for the suggestions, added! 

> There might be some more chances for optimizations for PLR=0

Right, currently addressing the low hanging fruits.. :sweat_smile:

Reply:
Thanks for your quick update.
Looks good.